File: WebAPI/Kingsmen.Domain/Entities/FunctionalAbility.cs
using System.Collections.Generic;

namespace Kingsmen.Domain.Entities
{
    public class FunctionalAbility : Entity
    {
        public FunctionalAbility() { }

        public string Name { get; set; }
        public string Description { get; set; }

        public List<FunctionalAbilityControlPoint> FunctionalAbilityControlPoints { get; set; }
        public List<RoleFunctionalAbility> RoleFunctionalAbilities { get; set; }
    }
}

File: React_UI/ClientApp/src/models/entities/FunctionalAbility.ts
import { IEntity } from "../Interfaces/IEntity";
import { FunctionalAbilityControlPoint } from "./FunctionalAbilityControlPoint";

export interface FunctionalAbility extends IEntity{
  name: string;
  description: string;
  createDateTime: Date;
  updateDateTime: Date;
  functionalAbilityControlPoints: Array<FunctionalAbilityControlPoint>;
}
